src/realm/index_string.cpp

```diff
@@ -930,39 +930,38 @@ void StringIndex::insert_row_list(size_t ref, size_t offset, StringData index_da
     m_array->insert(ins_pos + 1, ref);
 }
 
-
-void StringIndex::TreeInsert(ObjKey obj_key, key_type key, size_t offset, StringData index_data, const Mixed& value)
+void StringIndex::new_node(const NodeChange& nc)
 {
-    NodeChange nc = do_insert(obj_key, key, offset, index_data, value);
+    StringIndex new_node(inner_node_tag(), m_array->get_alloc());
     switch (nc.type) {
         case NodeChange::change_None:
-            return;
+            break;
```

Comment:
It appears that this case should never happen, should we have a REALM_UNREACHABLE here?